test/e2e/page-objects/pages/home/homepage.ts

```diff
@@ -269,30 +272,97 @@ class HomePage {
+      );
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Helper method to check if the empty state component is displayed.
+   *
+   * @returns Promise that resolves to true if empty state is found, false otherwise.
+   */
+  private async hasEmptyState(): Promise<boolean> {
+    try {
+      await this.driver.waitForSelector(this.emptyBalance, { timeout: 2000 });
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Helper method to get the current balance text for error messages.
+   *
+   * @returns Promise that resolves to the balance text, or 'unknown' if not found.
+   */
+  private async getActualBalance(): Promise<string> {
+    try {
+      const balanceElement = await this.driver.waitForSelector(this.balance, {
+        timeout: 2000,
+      });
+      return await balanceElement.getText();
+    } catch {
+      return 'unknown';
+    }
+  }
+
   /**
    * Checks if the expected balance is displayed on homepage.
    *
+   * This method handles three scenarios:
+   * 1. Regular balance display showing the expected value
+   * 2. Empty state component (valid only when expecting $0)
+   * 3. Balance mismatch - provides clear error with actual vs expected
+   *
    * @param expectedBalance - The expected balance to be displayed. Defaults to '25'.
    * @param symbol - The symbol of the currency or token. Defaults to 'ETH'.
    */
   async checkExpectedBalanceIsDisplayed(
     expectedBalance: string = '25',
     symbol: string = 'ETH',
   ): Promise<void> {
-    try {
-      await this.driver.waitForSelector({
-        css: this.balance,
-        text: expectedBalance,
-      });
-    } catch (e) {
-      const balance = await this.driver.waitForSelector(this.balance);
-      const currentBalance = parseFloat(await balance.getText());
-      const errorMessage = `Expected balance ${expectedBalance} ${symbol}, got balance ${currentBalance} ${symbol}`;
-      console.log(errorMessage, e);
-      throw e;
+    // Try 1: Check for regular balance display
+    if (await this.hasBalanceWithValue(expectedBalance)) {
+      console.log(
+        `Expected balance ${expectedBalance} ${symbol} is displayed on homepage`,
+      );
+      return;
     }
-    console.log(
-      `Expected balance ${expectedBalance} ${symbol} is displayed on homepage`,
+
+    // Try 2: Check for empty state (valid only for $0)
+    if (await this.hasEmptyState()) {
+      if (expectedBalance === '0') {
+        console.log(
+          'Balance empty state is displayed as expected for 0 balance',
+        );
+        return;
+      }
+      throw new Error(
+        `Expected balance ${expectedBalance} ${symbol}, but balance empty state is displayed instead`,
+      );
+    }
+
+    // Try 3: Get actual balance for error message
+    const actualBalance = await this.getActualBalance();
+    throw new Error(
+      `Expected balance ${expectedBalance} ${symbol}, got balance ${actualBalance} ${symbol}`,
```

Comment:
Refactoring to check for the balance empty state and opted to use helpers as opposed to many nested try catches for readability